### PR TITLE
[spark] Fix Ray on Spark fractional GPU error

### LINKED ISSUE #39537: Ray on Spark fractional GPU error calling setup_ray_cluster
### What happened + What you expected to happen

1. Bug spark.task.resource.gpu.amount does not support fractional GPU which is required for parallel spark jobs on GPU
https://github.com/ray-project/ray/blob/master/python/ray/util/spark/cluster_init.py
line 1026
num_spark_task_gpus = int(
  spark.sparkContext.getConf().get("spark.task.resource.gpu.amount", "0")
)

2. Ignore spark.task.resource.gpu.amount if num_spark_task_gpus passed manually

### Versions / Dependencies

All versions

### Reproduction script

Set spark.task.resource.gpu.amount to fractional value and num_gpus_worker_node to not None in call to setup_ray_cluster

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Why are these changes needed?

Fix error reported in https://github.com/ray-project/ray/issues/39537

## Related issue number

Closes #39537

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @jjyao

File: python/ray/util/spark/cluster_init.py [hunk 1]
```diff
@@ -1022,8 +1022,9 @@ def _setup_ray_cluster_internal(
     if num_cpus_worker_node is not None and num_cpus_worker_node <= 0:
         raise ValueError("Argument `num_cpus_worker_node` value must be > 0.")
 
+    # note: spark.task.resource.gpu.amount config might be fractional value like 0.5
     num_spark_task_gpus = int(
-        spark.sparkContext.getConf().get("spark.task.resource.gpu.amount", "0")
+        float(spark.sparkContext.getConf().get("spark.task.resource.gpu.amount", "0"))
```
Comment (jjyao) [inline python/ray/util/spark/cluster_init.py:1027]: We should raise an error if `"spark.task.resource.gpu.amount"` is float given ray node doesn't support fractional gpu?

Comment (WeichenXu123) [reply python/ray/util/spark/cluster_init.py]: Discussed offline. No need to raise error, instead, we can document the behavior.

File: python/ray/util/spark/cluster_init.py [hunk 2]
```diff
@@ -1084,9 +1086,9 @@ def _get_spark_worker_resources(_):
                 "`num_gpus_worker_node` values. Without setting the 2 arguments, "
                 "per-Ray worker node will be assigned with number of "
                 f"'spark.task.cpus' (equals to {num_spark_task_cpus}) cpu cores "
-                "and number of 'spark.task.resource.gpu.amount' "
-                f"(equals to {num_spark_task_gpus}) GPUs. To enable spark stage "
-                "scheduling, you need to upgrade spark to 3.4 version or use "
+                "and down-rounded number of 'spark.task.resource.gpu.amount' "
```
Comment (jjyao) [inline python/ray/util/spark/cluster_init.py:1089]:
```suggestion
                "and rounded down number of 'spark.task.resource.gpu.amount' "
```

File: python/ray/util/spark/cluster_init.py [hunk 3]
```diff
@@ -1377,7 +1379,7 @@ def setup_ray_cluster(
         num_gpus_worker_node: Number of gpus available to per-ray worker node, if not
             provided, if spark stage scheduling is supported, 'num_gpus_worker_node'
             value equals to number of GPUs per spark worker node, otherwise
-            it uses spark application configuration
+            it uses down-rounded value of spark application configuration
```
Comment (jjyao) [inline python/ray/util/spark/cluster_init.py:1382]:
```suggestion
            it uses rounded down value of spark application configuration
```

**Review by jjyao — APPROVED** [review]

**Comment by WeichenXu123** [comment]
@jjyao Done!

**Pull request merged**